src/coreclr-debug/debugConfigurationProvider.ts

```diff
@@ -59,7 +63,64 @@ export class DotnetDebugConfigurationProvider implements vscode.DebugConfigurati
                 return undefined;
             }
         }
+        
+        // We want to ask the user if we should run dotnet  dev-certs https --trust, but this doesn't work in a few cases --
+        // Linux -- not supported by the .NET CLI as there isn't a single root cert store
+        // VS Code remoting/Web UI -- the trusted cert work would need to happen on the client machine, but we don't have a way to run code there currently
+        // pipeTransport -- the dev cert on the server will be different from the client
+        if (!this.platformInformation.isLinux() && !vscode.env.remoteName && vscode.env.uiKind != vscode.UIKind.Web && !debugConfiguration.pipeTransport)
+        {
+            if(debugConfiguration.checkForDevCert === undefined && debugConfiguration.serverReadyAction)
+            {
+                debugConfiguration.checkForDevCert = true;
+            }
+
+            if (debugConfiguration.checkForDevCert)
+            {
+                checkForDevCerts(this.options.dotNetCliPaths, this.eventStream);
+            }
+        }
 
         return debugConfiguration;
     }
 }
+
+function checkForDevCerts(dotNetCliPaths: string[], eventStream: EventStream){
+    hasDotnetDevCertsHttps(dotNetCliPaths).then(async (returnData) => {
+        if(returnData.error) //if the prcess returns 0 error is null, otherwise the return code can ba acessed in returnData.error.code
+        {
+            const labelYes: string = "Yes";
+            const labelNotNow: string = "Not Now";
+            const labelMoreInfo: string = "More Information";
+
+            const result = await vscode.window.showInformationMessage(
+                "The selected launch configuration is configured to launch a web browser but no trusted development certificate was found. Create a trusted self-signed certificate?", 
+                { title:labelYes }, { title:labelNotNow, isCloseAffordance: true }, { title:labelMoreInfo }
+                ); 
+            if (result?.title === labelYes)
+            {
+                let returnData = await createSelfSignedCert(dotNetCliPaths);
+                if (returnData.error === null)
+                {
+                    vscode.window.showInformationMessage('Self-signed certificate sucessfully created.');
+                }
+                else
+                {
+                    eventStream.post(new DevCertCreationFailure(`${returnData.error.message}\ncode: ${returnData.error.code}\nstdout: ${returnData.stdout}`));
+
+                    const labelShowOutput: string = "Show Output";
+                    const result = await vscode.window.showWarningMessage("Couldn't create self-signed certificate. See output for more information.", labelShowOutput);
+                    if (result === labelShowOutput){
+                        vscode.commands.executeCommand("workbench.action.output.show.extension-output-ms-dotnettools.csharp-#3-C#");
```

Comment:
>workbench.action.output.show.extension-output-ms-dotnettools.csharp-#3-C# [](http://example.com/codeflow?start=56&length=73)

I feel like this command name is not stable.

Reply:
I think you want to use `.show` on the output channel: [https://github.com/microsoft/vscode/blob/c83eff40dd2c2115f5fd5e67a6ce304fbc6fdefa/src/vscode-dts/vscode.d.ts#L6448](https://github.com/microsoft/vscode/blob/c83eff40dd2c2115f5fd5e67a6ce304fbc6fdefa/src/vscode-dts/vscode.d.ts#L6448)